hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/BlockingRpcConnection.java

```diff
@@ -253,7 +252,17 @@ protected void setupConnection() throws IOException {
         if (this.rpcClient.localAddr != null) {
           this.socket.bind(this.rpcClient.localAddr);
         }
-        NetUtils.connect(this.socket, remoteId.getAddress(), this.rpcClient.connectTO);
+        if (this.rpcClient.metrics != null) {
```

Comment:
This change fixes a bug we encountered in production while running in Amazon's Elastic Kubernetes Service.